### DISCUSSION TITLE
Support of Mapbox Raster Tile in raster-array format

### DISCUSSION THREAD

**Discussion by zy6p** [developmentseed/titiler#810] [Q&A] [answered]
I propose adding support for Mapbox's beta [raster-array](https://docs.mapbox.com/mapbox-tiling-service/guides/raster-mts/) format for raster tiles, which offers significant compression and allows dynamic styling capabilities on the frontend(implement in [mapbox-gl-js](https://github.com/mapbox/mapbox-gl-js/releases/tag/v3.3.0)). This format enables efficient parsing and image restoration, significantly reducing bandwidth usage and improving load times.

Integrating this feature will enhance interactive map experiences by allowing client-side style adjustments without intensive server-side processing.

Thank you for considering this integration to improve functionality and user interaction.

**Comment by vincentsarago** [comment]
@zy6p I'm not quite sure to understand what is needed on titiler side

**Comment by zy6p** [comment]
@vincentsarago Can titiler support output in Mapbox Raster Tile raster-array format?

**Reply by vincentsarago** [reply]
> Can titiler support output in Mapbox Raster Tile raster-array format?

Sure but someone has to implement it 😅

**Reply by vincentsarago** [reply]
oh and I don't think the `MRT` format is open source 🤷

**Reply by zy6p** [reply]
It is indeed not open source; currently, only [the parsing logic](https://github.com/mapbox/mapbox-gl-js/blob/main/src/data/mrt/mrt.js) is open-sourced. It might require reverse engineering.👀

**Reply by vincentsarago** [reply, marked as answer]
I would be really careful reverse-engineer anything from Mapbox, especially by looking at mapbox-gl-js. Their legal team could not like this!

**Reply by zy6p** [reply]
Thanks for the caution! I’ll look into alternative solutions that comply with open-source standards. 😊
